EKrut-Server/src/ekrut/server/db/ReportDAO.java

```diff
@@ -350,8 +387,133 @@ public Object fetchAllMonthlyOrders(LocalDateTime date, String location) {
 				} catch (SQLException e) {
 					throw new RuntimeException(e);
 				}
+			}
+	}
+	
+	public ArrayList<String> getThresholdAlert(LocalDateTime date, String ekrutLocation){
+		
+		PreparedStatement ps1 = con.getPreparedStatement("SELECT itemName FROM thresholdAlerts WHERE date = ? AND location = ?");
+		try {
+			ps1.setObject(1, date, MysqlType.DATETIME);
+			ps1.setString(2, ekrutLocation);
+			
+			ResultSet rs1 = con.executeQuery(ps1);
+			
+			ArrayList<String> thresholdAlerts = new ArrayList<>();	
+			
+			while (rs1.next()) {
+				thresholdAlerts.add(rs1.getString("itemName"));
+			}
+			
+			return thresholdAlerts;
+			
+			}catch (SQLException e) {
+				return null;
+			} finally { 
+				try {
+					ps1.close();
+				} catch (SQLException e) {
+					throw new RuntimeException(e);
+				}
+			}
+	}
+	
+	
+	public boolean createReport(Report report) {
+		con.beginTransaction();
+		
+		// Pass true so that we can get the new report ID
+		PreparedStatement p1 = con.getPreparedStatement("INSERT INTO reports " +
+                                                        "(type,date,location) " +
+                                                        "VALUES(?,?,?)", true);
+		
+		try {
+			p1.setString(1, report.getReportType().toString());
+			p1.setObject(2, report.getDate(), MysqlType.DATETIME);
+			p1.setString(3, report.getEkrutLocation().toString());
+	
+			// Get the new report ID
+			ResultSet rs = p1.getGeneratedKeys();
+			if (!rs.next()) {
+				con.abortTransaction();
+				return false;
+			}
+			
+			int reportID = rs.getInt(1);
+			rs.close();
+			
+			// Save the ID in the report object for later use
+			report.setReportID(reportID);
+			
+			con.commitTransaction();
+			
+			return true;
 			
+		} catch (SQLException e) {
+			con.abortTransaction();
+			return false;
+		} finally {
+			try {
+				p1.close();
+			} catch (SQLException e) {
+				throw new RuntimeException(e);
+			}
+		}
 	}
+	
+	public boolean createOrderReport(Report report) {
+		
+		con.beginTransaction();
+		/* * orders monthly sales table:
+ *   reportID  | totalOrders | totalOrdersInILS |
+ * --------------------------------------------
+ * 	  15678    |    3220    |       18550     |*/
+		
+		
+		PreparedStatement ps1 = con.getPreparedStatement("INSERT INTO orderMonthlySales"
```

Comment:
You're not using `ps1` here for anything. Also, the formatting of these lines is a bit broken.